ui/packages/catalog-ui-search/src/main/webapp/react-component/container/sharing/sharing.tsx

```diff
@@ -128,39 +128,58 @@ export class Sharing extends React.Component<Props, State> {
     const loadingView = new LoadingView()
     this.attemptSave(attributes, usersToUnsubscribe)
       .then(() => {
-        this.showSaveSuccessful()
+        announcement.announce(
+          {
+            title: 'Success',
+            message: 'Sharing saved',
+            type: 'success',
+          },
+          1500
+        )
         this.props.lightbox.close()
       })
       .catch(err => {
-        if (err.message === 'Need to refresh') {
-          this.showNeedToRefresh()
+        if (err.message === 'concurrent-modification') {
+          announcement.announce(
+            {
+              title: 'The workspace settings could not be updated',
+              message:
+                'The workspace has been modified by another user. Please refresh the page and reattempt your changes.',
+              type: 'error',
+            },
+            1500
+          )
         } else {
-          this.showSaveFailed()
+          announcement.announce(
+            {
+              title: 'Error',
+              message: 'Save failed',
+              type: 'error',
+            },
+            1500
+          )
         }
       })
       .then(() => {
         loadingView.remove()
       })
   }
 
-  // NOTE: Fetching the latest workspace and checking the modified dates is a temporary solution
+  // NOTE: Fetching the latest metacard and checking the modified dates is a temporary solution
   // and should be removed when support for optimistic concurrency is added
   // https://github.com/codice/ddf/issues/4467
   attemptSave = async (attributes: any, usersToUnsubscribe: String[]) => {
     const currWorkspace = await this.fetchWorkspace(this.props.id)
-    if (
-      currWorkspace['metacard.modified'] ===
-      this.state.previousWorkspace['metacard.modified']
-    ) {
+    if (currWorkspace['metacard.modified'] === this.state.modified) {
       await this.doSave(attributes)
       await this.unsubscribeUsers(usersToUnsubscribe)
       const newWorkspace = await this.fetchWorkspace(this.props.id)
```

Comment:
Should we update this to be `newMetacard` and the function to be `fetchMetacard` to keep in line with all the other changes being made?

Reply:
good call